content/en/blog/falcosidekick-reponse-engine-part-5-argo.md

```diff
@@ -0,0 +1,377 @@
+
+## Installation of Argo Events
+
+We simply follow the [official documentation](https://argoproj.github.io/argo-events/installation/).
+
+```shell+
+kubectl create namespace argo-events
+kubectl apply -n argo-events -f https://raw.githubusercontent.com/argoproj/argo-events/stable/manifests/install.yaml
+kubectl apply -n argo-events -f https://raw.githubusercontent.com/argoproj/argo-events/stable/manifests/install-validating-webhook.yaml
+kubectl apply -n argo-events -f https://raw.githubusercontent.com/argoproj/argo-events/stable/examples/eventbus/native.yaml
+```
+
+## Installation of Argo Workflow
+
+Again, the [official documentation](https://argoproj.github.io/argo-workflows/installation/) will help us.argo
+
+```shell+
+kubectl create namespace argo
+kubectl apply -n argo -f https://raw.githubusercontent.com/argoproj/argo-workflows/stable/manifests/install.yaml
+kubectl patch -n argo cm workflow-controller-configmap -p '{"data": {"containerRuntimeExecutor": "pns"}}'
+```
+
+The `kubectl patch` is there for allowing the workflows to run in `minikube`, `kind`, etc. See [docs](https://argoproj.github.io/argo-workflows/workflow-executors/) about Workflow Executors.
+
+After a while, you should have access to `Argo Workflow` UI through a port-forward:
+
+```shell+
+kubectl port-forward svc/argo-server -n argo 2746:2746 &
+```
+
+The link is [https://localhost:2746](https://localhost:2746) (you can ignore the certificate error, we're in a lab 😉).
+
+## Creation of the Event Source 
+
+We'll use an `Event Source` with `Webhook` type. It will receive `Falco` events from `Falcosidekick` and push them then into the Event Bus.
+
+This component is pretty easy to understand. `Falcosidekick` will have to **POST** the events to an endpoint **/falco** of a service opened on port **12000**. *Easy*.
+
+```yaml+
+cat <<EOF | kubectl apply -n argo-events -f -
+apiVersion: argoproj.io/v1alpha1
+kind: EventSource
+metadata:
+  name: webhook-falco
+  namespace: argo-events
+spec:
+  service:
+    ports:
+      - port: 12000
+        targetPort: 12000
+  webhook:
+    # event-source can run multiple HTTP servers. Simply define a unique port to start a new HTTP server
+    falco-event:
+      # port to run HTTP server on
+      port: "12000"
+      # endpoint to listen to
+      endpoint: /falco
+      # HTTP request method to allow. In this case, only POST requests are accepted
+      method: POST
+EOF
+```
+
+As expected, we do have a new service which will listen events from `Falcosidekick` on port **12000**:
+
+```shell+
+kubetcl get svc -n argo-events
+
+NAME                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+webhook-falco-eventsource-svc   ClusterIP   10.43.117.26    <none>        12000/TCP                    11m
+```
+
+## Creation of the Sensor
+
+In `Argo Events` architecture, `Sensors` are responsible to listen the Event Bus and triggers *something* if the criterias we set match. 
```

Comment:
'Sensors' are responsible for listening to the Event Bus and triggering *somerthing should the criteria we set match